frontend/src/component/feature/FeatureView/FeatureOverview/ReleasePlan/ReleasePlanAddChangeRequestDialog.tsx

```diff
@@ -0,0 +1,62 @@
+import { Dialogue } from 'component/common/Dialogue/Dialogue';
+import { useReleasePlanTemplate } from 'hooks/api/getters/useReleasePlanTemplates/useReleasePlanTemplate';
+import useToast from 'hooks/useToast';
+import { styled, Button } from '@mui/material';
+
+const StyledBoldSpan = styled('span')(({ theme }) => ({
+    fontWeight: theme.typography.fontWeightBold,
+}));
+
+export const ReleasePlanAddChangeRequestDialog = ({
+    featureId,
+    environmentId,
+    releaseTemplateId,
+    onClosing,
+}: {
+    featureId: string;
+    environmentId: string;
+    releaseTemplateId: string | undefined;
+    onClosing: () => void;
+}) => {
+    if (!releaseTemplateId) {
+        return null;
+    }
+    const template = useReleasePlanTemplate(releaseTemplateId);
+    const { setToastData } = useToast();
+
+    const addReleasePlanToChangeRequest = async () => {
+        setToastData({
+            type: 'success',
+            text: 'Added to draft',
+        });
+        onClosing();
+    };
+
+    return (
+        <Dialogue
+            title='Request changes'
+            open={true}
```

Comment:
Maybe it would be more consistent to manage the open state from the caller, like we do for other dialogs?

Reply:
Even if it's something like open={Boolean(changeRequestReleaseTemplateId)}, since that seems to be your logic here.

Reply:
Yeah thanks, I agree and tried that, but there's a call to get template that needs a defined string. So I ended up getting a runtime crash due to number of hooks being rendered. I might be able to get out of that by changing the type of what goes into the templateId state to hold all I need instead

Reply:
[This](https://github.com/Unleash/unleash/pull/9139/files#diff-5a7b0924651d248db4f47c573a7e8c907f71d6c7f4de84db7193c9fe569619f1R24) one? You already have a guard above so you shouldn't have that issue anymore.

What I meant is that `open` would be a prop in this component (`ReleasePlanAddChangeRequestDialog`, so here you would simply have `open={open}`. But maybe the caller decides that `open={Boolean(changeRequestReleaseTemplateId)}` if you don't want to manage a separate open state and instead just rely on that ID being set.

Reply:
>I might be able to get out of that by changing the type of what goes into the templateId state to hold all I need instead

I like this one as well, you can simply pass the template itself if you have it instead.